src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -5464,6 +5505,63 @@ static void validate(ModuleSymbol module)
             }
         }
 
+        [Fact]
+        public void StaticLocalFunction_ConditionalAttribute_Unreferenced()
+        {
+            var source = @"
+using System.Diagnostics;
+using System;
+
+class C
+{
+    static void Main()
+    {
+        local1();
+
+        [Conditional(""DEBUG"")]
+        static void local1()
+        {
+            Console.Write(""hello"");
+        }
+    }
+}
+";
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview).VerifyDiagnostics(
+                // (12,21): warning CS8321: The local function 'local1' is declared but never used
+                //         static void local1()
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "local1").WithArguments("local1").WithLocation(12, 21));
+
+            CreateCompilation(source, parseOptions: TestOptions.RegularPreview.WithPreprocessorSymbols("DEBUG")).VerifyDiagnostics();
```

Comment:
I was slightly surprised by this behavior but I don't see a need to change it right now.